File: algorithms/divide-and-conquer/decimal-to-binary.ts
/**
 * Dezimalkonverter
 *
 * Konvertiert eine Dezimalzahl in eine Binärzahl.
 *
 * Komplexität: O(log n)
 */

const decimalToBinary = (value: number): string => {
  if (value < 1) {
    return "";
  }
  return decimalToBinary(Math.floor(value / 2)) + (value % 2);
};

const result = decimalToBinary(182);
console.log(`Binary: ${result}`);
export {};

File: algorithms/divide-and-conquer/decimal-to-binary.py
"""
Dezimalkonverter

Konvertiert eine Dezimalzahl in eine Binärzahl.

Komplexität: O(log n)
"""


def decimalToBinary(decimal: int):
    if decimal < 1:
        return 0
    else:
        decimalToBinary(int(decimal/2))
        print(decimal % 2)


if __name__ == "__main__":
    decimalToBinary(9)
